src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,170 @@
+      } else {
+        this.addClass("horizontal");
+      }
+
+      this._dragInteraction = new Plottable.Interactions.Drag();
+      this._dragInteraction.attachTo(this);
+
+      let grabbedLine = (p: Point) => {
+        return (this._isVertical() &&
+                 p.x >= this.pixelPosition() - this.detectionRadius() &&
+                 p.x <= this.pixelPosition() + this.detectionRadius()
+               ) ||
+               (
+                 !this._isVertical() &&
+                 p.y >= this.pixelPosition() - this.detectionRadius() &&
+                 p.y <= this.pixelPosition() + this.detectionRadius()
+               );
+      };
+
+      let dragging = false;
+      let dragStartCallback = (start: Point) => {
+        if (grabbedLine(start)) {
+          dragging = true;
+          this._dragStartCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragStart(dragStartCallback);
+      let dragCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          this._setPixelPositionWithoutChangingMode(this._isVertical() ? end.x : end.y);
+          this._dragCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDrag(dragCallback);
+      let dragEndCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          dragging = false;
+          this._dragEndCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragEnd(dragEndCallback);
+
+      this._disconnectInteractionCallbacks = () => {
+        this._dragInteraction.offDragStart(dragStartCallback);
+        this._dragInteraction.offDrag(dragCallback);
+        this._dragInteraction.offDragEnd(dragEndCallback);
+        delete this._disconnectInteractionCallbacks;
+      };
+
+      this._dragStartCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragEndCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+    }
+
+    protected _setup() {
+      super._setup();
+      this._detectionEdge = this.content().append("line").style({
+                              opacity: 0,
+                              fill: "pink",
+                              "pointer-events": "visibleStroke"
+                            }).classed("drag-edge", true);
+    }
+
+    public renderImmediately() {
+      super.renderImmediately();
+      this._detectionEdge.attr({
+        x1: this._isVertical() ? this.pixelPosition() : 0,
+        y1: this._isVertical() ? 0 : this.pixelPosition(),
+        x2: this._isVertical() ? this.pixelPosition() : this.width(),
+        y2: this._isVertical() ? this.height() : this.pixelPosition(),
+        "stroke-width": this._detectionRadius * 2
+      });
+
+      return this;
+    }
+
+    public detectionRadius(): number;
+    public detectionRadius(detectionRadius: number): DragLineLayer<D>;
+    public detectionRadius(detectionRadius?: number): any {
```

Comment:
Maybe name endpoint to `detectionWidth()`?  Radius sounds misleading since it's a line not a cylinder or anything circular